### PR TITLE
Hello rails react

### PR DESCRIPTION
**Pull request by luis-pomare**
In this milestone, we have:
- Added all migration files and database modifications to have a fully functional model
- Created the API endpoint that responds with a random greeting
- Added all necessary configurations to use redux to manage the app state
- Added all necessary configurations to use react router

### CODE REVIEW

File: app/javascript/application.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { Route, Routes, BrowserRouter } from 'react-router-dom';
+
+ReactDOM.render(
+  <Provider store={store}>
+    <BrowserRouter>
+      <Routes>
+        <Route path="/" element={<App />} />
+        <Route path="/greeting" element={<App />} />
+      </Routes>
+    </BrowserRouter>
```
Comment (Ismailco) [inline app/javascript/application.js]: - [x] Kindly make sure to move this code into the `App` component instead as per requirment the App component chould have the Routes

```suggestion
    <App />
```

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  return (
+    <h1>
+      {greeting}
+    </h1>
+  );
+}
+
+const mapStateToProps = (state) => ({
+  greeting: state.greeting.text,
```
Comment (Ismailco) [inline app/javascript/components/App.js]: - [x] This Component `App` should have the routes, kindly make sure to create a new component named `Greeting` and move this code to it then replace the App content with the routes from the `application.js`

```suggestion
      <BrowserRouter>
      <Routes>
        <Route path="/" element={<App />} />
        <Route path="/greeting" element={<App />} />
      </Routes>
    </BrowserRouter>
```

Comment (Ismailco) [inline app/javascript/components/App.js]: - [x] This Component `App` should have the routes, kindly make sure to create a new component named `Greeting` and move this code to it then replace the App content with the routes from the `application.js`

```suggestion
      <BrowserRouter>
      <Routes>
        <Route path="/" element={<App />} />
        <Route path="/greeting" element={<App />} />
      </Routes>
    </BrowserRouter>
```

File: app/javascript/components/App.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from 'react';
+import { connect, useDispatch, useSelector } from 'react-redux';
+import { setGreeting } from '../actions/greetingActions';
+
+function App() {
```
Comment (Ismailco) [inline app/javascript/components/App.js:5]: - [x] Kindly make sure to create a new component `Greeting` with the content of this component

```javascript
  const greeting = useSelector(state => state.greeting.text);
  const dispatch = useDispatch();

  useEffect(() => {
    fetch('/greeting/api')
      .then(response => response.json())
      .then(data => {
        dispatch(setGreeting(data.text));
      })
      .catch(error => {
        console.error(error);
      });
  }, [dispatch]);

  return (
    <h1>
      {greeting}
    </h1>
  );
}

const mapStateToProps = (state) => ({
  greeting: state.greeting.text,
```

File: app/javascript/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// reducers/index.js
+import { combineReducers } from 'redux';
+import greetingReducer from './greetingReducer';
+
+const rootReducer = combineReducers({
```
Comment (Ismailco) [inline app/javascript/reducers/index.js]: - [x] [OPTIONAL] I would suggest creating a folder named `redux` and moving the index.js and reducers folder inside. The folder structure can be like this 👇🏼 
```
...
...
- redux
  -- index.js
  -- reducers
    --- greetingReducer.js
...
...
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @luis-pomare, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Luis,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾Congratulations!  🎉 💯 🌟 

## Highlights 🎯 
🎲  Connection between Rails back-end and React front-end properly established in one app!

Cheers..Happy coding! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/App.js (hunk 3 above)
Comment (rloterh) [reply app/javascript/components/App.js:5]: well resolved! ✔️

File: app/javascript/components/App.js (hunk 2 above)
Comment (rloterh) [reply app/javascript/components/App.js]: well resolved! ✔️

File: app/javascript/application.js (hunk 1 above)
Comment (rloterh) [reply app/javascript/application.js]: well resolved! ✔️

File: app/javascript/components/App.js (hunk 2 above)
Comment (rloterh) [reply app/javascript/components/App.js]: well resolved! ✔️

**Pull request merged**